src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,49 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type != null) {
+            if (type == Bulldozer.class) {
+                MachineProducer<Bulldozer> bulldozeeProducer = new BulldozerProducer();
+                return new ArrayList<>(bulldozeeProducer.get());
+            }
+            if (type == Excavator.class) {
+                MachineProducer<Excavator> excavatorProducer = new ExcavatorProducer();
+                return new ArrayList<>(excavatorProducer.get());
+            }
+            if (type == Truck.class) {
+                MachineProducer<Truck> truckProducer = new TruckProducer();
+                return new ArrayList<>(truckProducer.get());
+            }
+        }
+        return new ArrayList<>();
+    }
```

Comment:
```suggestion
public class MachineServiceImpl implements MachineService<Machine> {
    private final MachineProducer<Bulldozer> bulldozerProducer = new BulldozerProducer;
    private final MachineProducer<Excavator> excavatorProducer = new ExcavatorProducer();
    private final MachineProducer<Truck> truckProducer = new TruckProducer();
    
    @Override
    public List<Machine> getAll(Class<? extends Machine> type) {
        if (type != null) {
            if (type == Bulldozer.class) {
                return new ArrayList<>(bulldozeeProducer.get());
            }
            if (type == Excavator.class) {
                return new ArrayList<>(excavatorProducer.get());
            }
            if (type == Truck.class) {
                return new ArrayList<>(truckProducer.get());
            }
        }
        return new ArrayList<>();
    }
```
you can create producers just once